### PR TITLE
DOC: update data structure docs with links to classes/methods

### PR DESCRIPTION
**Pull request by snowman2**
Also added autodoc for `set_geometry` method as it was missing and needed for data sructure docs.

### CODE REVIEW

File: doc/source/data_structures.rst [hunk 1]
```diff
@@ -13,14 +13,14 @@
 Data Structures
 =========================================
 
-GeoPandas implements two main data structures, a ``GeoSeries`` and a
-``GeoDataFrame``.  These are subclasses of pandas ``Series`` and
+GeoPandas implements two main data structures, a :class:`~geopandas.GeoSeries` and a
```
Comment (jorisvandenbossche) [inline doc/source/data_structures.rst]: All the `~geopandas.`'s should not be needed (as geopandas is the "currentmodule)

Comment (snowman2) [reply doc/source/data_structures.rst]: I got rid of most of them, but left a couple so only the methods show up to be consistent with how the docs were previously.

File: doc/source/data_structures.rst [hunk 2]
```diff
@@ -55,39 +55,39 @@ in :doc:`geometric manipulations <geometric_manipulations>`.
 
 Attributes
 ^^^^^^^^^^^^^^^
-* ``area``: shape area (units of projection -- see :doc:`projections <projections>`)
-* ``bounds``: tuple of max and min coordinates on each axis for each shape
-* ``total_bounds``: tuple of max and min coordinates on each axis for entire GeoSeries
-* ``geom_type``: type of geometry.
-* ``is_valid``: tests if coordinates make a shape that is reasonable geometric shape (`according to this <http://www.opengeospatial.org/standards/sfa>`_).
+* :attr:`~geopandas.GeoSeries.area`: shape area (units of projection -- see :doc:`projections <projections>`)
```
Comment (jorisvandenbossche) [inline doc/source/data_structures.rst]: BTW, you can still remove the `geopandas.` part. If you don't want the GeoSeries to show up, only the `~` is needed.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**